graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -460,187 +429,224 @@ def _split_classes_by_kind(self, class_name_to_definition):
+                                                        kind, inheritance_set,
+                                                        self._non_graph_class_names)
+            # Perform validation.
+            _validate_links_not_defined_on_non_edge_class(class_name, kind, links)
 
-            from_class_name = edge_element.properties[EDGE_SOURCE_PROPERTY_NAME].qualifier
-            to_class_name = edge_element.properties[EDGE_DESTINATION_PROPERTY_NAME].qualifier
+            # Store.
+            self._elements[class_name] = elem
 
-            edge_schema_element = self._elements[edge_class_name]
+        # Set vertex connections.
+        for edge_class_name in self._edge_class_names:
+            edge = self._elements[edge_class_name]
+            if edge.abstract:
+                continue
+            from_class_name = get_only_element_from_collection(edge.in_connections)
+            to_class_name = get_only_element_from_collection(edge.out_connections)
 
-            # Link from_class_name with edge_class_name
             for from_class in self._subclass_sets[from_class_name]:
                 from_schema_element = self._elements[from_class]
                 from_schema_element.out_connections.add(edge_class_name)
-                edge_schema_element.in_connections.add(from_class)
 
-            # Link edge_class_name with to_class_name
             for to_class in self._subclass_sets[to_class_name]:
                 to_schema_element = self._elements[to_class]
-                edge_schema_element.out_connections.add(to_class)
                 to_schema_element.in_connections.add(edge_class_name)
+
+        # Freeze connections.
+        for class_name in self.vertex_class_names:
+            self._elements[class_name].freeze()
+
+
+def _get_schema_element_and_links(class_name, class_name_to_definition, kind, inheritance_set,
+                                  non_graph_class_names):
+    """Return the schema element and its links. Only Edges have links."""
+    class_definition = class_name_to_definition[class_name]
+    class_fields = class_definition.get('customFields')
+    if class_fields is None:
+        # OrientDB likes to make empty collections be None instead.
+        # We convert this field back to an empty dict, for our general sanity.
+        class_fields = dict()
+    property_name_to_descriptor, links = (
+        _get_class_properties_and_links(class_name, class_name_to_definition, inheritance_set,
+                                        non_graph_class_names))
+    elem = SchemaElement(class_name, kind, _is_abstract(class_definition),
+                         property_name_to_descriptor, class_fields)
+    return elem, links
+
+
+def _get_class_properties_and_links(class_name, class_name_to_definition, inheritance_set,
+                                    non_graph_class_names):
+    """Return the properties descriptors and links of the class. Only Edges have links."""
```

Comment:
typo: `properties descriptors`. Also, `Edges` being capitalized makes me feel like it's a type, when I don't believe it is.